includes/ical-parser/class-coblocks-ical.php

```diff
@@ -0,0 +1,2782 @@
+												$an_event            = $this->process_event_ical_datetime( $an_event );
+												$recurrence_events[] = $an_event;
+												$this->event_count++;
+
+												// If RRULE[COUNT] is reached then break
+												if ( isset( $rrules['COUNT'] ) ) {
+													$count_nb++;
+
+													if ( $count_nb >= $count_orig ) {
+														break 2;
+													}
+												}
+											}
+										}
+									}
+
+									// Move forwards
+									$recurring_timestamp = strtotime( $offset, $recurring_timestamp );
+								}
+							} elseif ( isset( $rrules['BYDAY'] ) && '' !== $rrules['BYDAY'] ) {
+								while ( $recurring_timestamp <= $until ) {
+									$month_recurring_timestamp = $recurring_timestamp;
+
+									// Adjust time zone from initial event
+									$monthrecurring_offset = 0;
+
+									if ( $this->use_timezone_with_r_rules ) {
+										$recurring_timezone = \DateTime::createFromFormat( self::UNIX_FORMAT, $month_recurring_timestamp );
+										$recurring_timezone->setTimezone( $initial_start->getTimezone() );
+										$monthrecurring_offset      = $recurring_timezone->getOffset();
+										$month_recurring_timestamp += $monthrecurring_offset;
+									}
+
+									$event_start_desc      = "{$this->convert_day_ordinal_to_positive($day_number, $weekday, $month_recurring_timestamp)} {$this->weekdays[$weekday]} "
+										. date( self::DATE_TIME_FORMAT_PRETTY, $month_recurring_timestamp );
+									$event_start_timestamp = strtotime( $event_start_desc );
+
+									if ( intval( $rrules['BYDAY'] ) === 0 ) {
+										$last_day_desc = "last {$this->weekdays[$weekday]} "
+											. date( self::DATE_TIME_FORMAT_PRETTY, $month_recurring_timestamp );
+									} else {
+										$last_day_desc = "{$this->convert_day_ordinal_to_positive($day_number, $weekday, $month_recurring_timestamp)} {$this->weekdays[$weekday]} "
+											. date( self::DATE_TIME_FORMAT_PRETTY, $month_recurring_timestamp );
+									}
+
+									$last_day_time_stamp = strtotime( $last_day_desc );
+
+									do {
+										// Prevent 5th day of a month from showing up on the next month
+										// If BYDAY and the event falls outside the current month, skip the event
+
+										$compare_current_month = date( 'F', $month_recurring_timestamp );
+										$compare_event_month   = date( 'F', $event_start_timestamp );
+
+										if ( $compare_current_month !== $compare_event_month ) {
+											$month_recurring_timestamp = strtotime( $offset, $month_recurring_timestamp );
+											continue;
+										}
+
+										if ( $event_start_timestamp > $start_timestamp && $event_start_timestamp <= $until ) {
+											$an_event['DTSTART']          = date( self::DATE_TIME_FORMAT, $event_start_timestamp ) . ( ( 'Z' === $initial_start_timezone_name ) ? 'Z' : '' );
+											$an_event['DTSTART_array'][1] = $an_event['DTSTART'];
+											$an_event['DTSTART_array'][2] = $event_start_timest
```

Comment:
In the context of WordPress we should be using [`WP_Error`](https://developer.wordpress.org/reference/classes/wp_error/) for error reporting.